scoring/src/main/scala/ai/h2o/sparkling/ml/metrics/H2OBinomialMetrics.scala

```diff
@@ -0,0 +1,159 @@
+    * @return Calculated binomial metrics
+    */
+  def calculate(
+      dataFrame: DataFrame,
+      domain: Array[String],
+      predictionCol: String = "detailed_prediction",
+      labelCol: String = "label",
+      weightColOption: Option[String] = None,
+      offsetColOption: Option[String] = None): H2OBinomialMetrics = {
+    validateDataFrameForMetricCalculation(dataFrame, domain, predictionCol, labelCol, offsetColOption, weightColOption)
+    val getMetricBuilder = () => new MetricBuilderBinomial(domain)
+    val castedLabelDF = dataFrame.withColumn(labelCol, col(labelCol) cast StringType)
+
+    val gson =
+      getMetricGson(getMetricBuilder, castedLabelDF, predictionCol, labelCol, offsetColOption, weightColOption, domain)
+    val result = new H2OBinomialMetrics()
+    result.setMetrics(gson, "H2OBinomialMetrics.calculate")
+    result
+  }
+
+  // The method serves for call from Python API
+  def calculateInternal(
+      dataFrame: DataFrame,
+      domain: java.util.ArrayList[String],
+      predictionCol: String,
+      labelCol: String,
+      weightCol: String,
+      offsetCol: String): H2OBinomialMetrics = {
+    calculate(
+      dataFrame,
+      domain.toArray[String](new Array[String](0)),
+      predictionCol,
+      labelCol,
+      Option(weightCol),
+      Option(offsetCol))
+  }
+
+  // The method serves for call from R API
+  def calculateInternal(
+      dataFrame: DataFrame,
+      domain: Array[String],
+      predictionCol: String,
+      labelCol: String,
+      weightCol: String,
+      offsetCol: String): H2OBinomialMetrics = {
+    calculate(dataFrame, domain, predictionCol, labelCol, Option(weightCol), Option(offsetCol))
+  }
+
+  override protected def getPredictionValues(dataType: DataType, domain: Array[String], row: Row): Array[Double] = {
+    dataType match {
+      case StructType(fields)
+          if fields(0).dataType == StringType && fields(1).dataType.isInstanceOf[StructType] &&
+            fields(1).dataType.asInstanceOf[StructType].fields.forall(_.dataType == DoubleType) &&
+            fields(1).dataType.asInstanceOf[StructType].fields.length == 2 =>
+        val predictionStructure = row.getStruct(0)
+        val prediction = predictionStructure.getString(0)
+        val index = domain.indexOf(prediction).toDouble
+        val probabilities = predictionStructure.getStruct(1)
+        Array(index) ++ probabilities.toSeq.map(_.asInstanceOf[Double])
+      case StructType(fields) if fields.forall(_.dataType == DoubleType) && fields.length == 2 =>
+        val probabilities = row.getStruct(0)
+        Array(-1.0) ++ probabilities.toSeq.map(_.asInstanceOf[Double])
+      case DoubleType => probabilityToArray(row.getDouble(0))
+      case FloatType => probabilityToArray(row.getFloat(0).toDouble)
+    }
+  }
+
+  private def probabilityToArray(probability: Double): Array[Double] = {
+    Array[Double](-1 /* unused */, 1 - probability, probability)
```

Comment:
if -1 means "unused" maybe it could be moved to a constant named for example "unusedValue" or something?

Reply:
will do, thank!